### PR TITLE
updated mintFunction-updateName

### LINKED ISSUE #24: Combine DAI transfers in mint()
Scott will be sending us code to accurately calculate cDAI amount minted, which will make this possible.

Will also allow us to use `saveDaiPriceInDaiCurrent` in `mint`

### LINKED ISSUE #30: Remove Ownable contract
Change the name of the token based on expiration date. Set oToken expiration date. If now is greater than expiration date, return name + "expired".

### PR DESCRIPTION
**Pull request by kseniya292**
closes #24 
closes #30

### CODE REVIEW

File: contracts/SaveDAI.sol [hunk 1]
```diff
@@ -108,38 +107,32 @@ contract SaveDAI is ISaveDAI, ERC20, ERC20Detailed, Pausable, Ownable {
...
-
-        // calculate how much DAI we need to pay to insure assetAmount
-        uint256 oTokenCost = getCostOfOToken(assetAmount);
+        // calculate DAI needed to buy _amount of ocDAI tokens
+        uint256 oTokenCost = getCostOfOToken(_amount);
 
-        // transfer DAI needed for premium for ocDAI tokens
-        require(dai.transferFrom(
+        // transfer total DAI needed
+        dai.transferFrom(
```
Comment (dougiebuckets) [inline contracts/SaveDAI.sol:119]: Let's wrap this in a `require` statement

File: contracts/SaveDAI.sol [hunk 2]
```diff
@@ -63,24 +62,24 @@ contract SaveDAI is ISaveDAI, ERC20, ERC20Detailed, Pausable, Ownable {
         ocDaiExchange = _getExchange(ocDaiAddress);
 
         require(
-            dai.approve(address(this), LARGE_APPROVAL_NUMBER) && 
+            dai.approve(address(this), LARGE_APPROVAL_NUMBER) &&
```
Comment (slgraham) [inline contracts/SaveDAI.sol:65]: Isn't this just giving ourselves approval to spend our own dai?

Comment (kseniya292) [reply contracts/SaveDAI.sol]: @dougiebuckets @slgraham Good point. I removed this, along with a couple other `approve` and `transferFrom` and replaced with `transfer` (when transferring from the saveDAI contract). Rather then approving and transferFrom, we can just directly transfer from our contract.

Comment (dougiebuckets) [reply contracts/SaveDAI.sol]: @slgraham good catch!

@kseniya292 - Lolo can't believe we missed that until now :)

**Pull request merged**